src/services/archanova.js

```diff
@@ -120,6 +120,51 @@ export const formatEstimated = (estimated: ?ParsedEstimate) => {
   };
 };
 
+const parseEstimateError = (error: Error, logExtras: Object = {}) => {
+  let errorMessage = t('error.unableToEstimateTransaction');
+  const errorReplaceString = '[ethjs-query] while formatting outputs from RPC'; // eslint-disable-line
+  if (error?.message) {
+    /**
+     * the return result from Archanova is problematic,
+     * it contains "ethjs-query" part and rest of message is JSON string
+     * this is known issue on Archanova end, but it's not planned to be fixed
+     * return example:
+     * [ethjs-query] while formatting outputs from RPC
+     * '{"value":{"body":"{\"oneMore\":\"escaped\",\"JSON\":\"object\"}"}}'
+     */
+    if (error.message.includes(errorReplaceString)) {
+      try {
```

Comment:
Definitely justified use of try/catch, when JSON.parse explodes - it explodes!

Reply:
FYI it's previous code ported as method to reuse on other new method.